dot/sync/chain_sync.go

```diff
@@ -176,7 +176,7 @@ type chainSyncConfig struct {
 	slotDuration       time.Duration
 }
 
-func newChainSync(cfg *chainSyncConfig) *chainSync {
```

Comment:
What's the motivation behind making `chainSyncConfig` from pointer to value?

Reply:
I had [this suggestion](https://github.com/ChainSafe/gossamer/pull/2906#discussion_r1009571677) for `func newChainProcessor(cfg *chainProcessorConfig) *chainProcessor`

> Also side-nit, if you do pass a settings struct, pass it as a value, not a pointer, there is really need no need to pass it as a pointer (no performance impact + fields are already interfaces + opens the door to pass nil)

Although I guess it applies in several places like here. Anyway imo those changes do not belong to this PR